src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -305,6 +312,30 @@ export class CellModel extends Disposable implements ICellModel {
 		this._preventNextChartCache = true;
 	}
 
+	private attachImageFromSource(newSource: string | string[]): string | string[] {
+		if (!Array.isArray(newSource) && this.isValidBase64OctetStream(newSource)) {
```

Comment:
Why are you ignoring array sources?  Seems like it would be fine to iterate through all the lines and do the replacement for each one. 

And if you do that then it seems like it'd be simpler to just have this take in an array only and put it after the `getMultilineSource` call. (which should also be updated to say it has a return type of string[] only since it can't actually return a string as it is currently.